src/test/e2e/helpers/database-init.ts

```diff
@@ -94,46 +96,79 @@ export interface ITestDb {
     rawDatabase: Knex;
 }
 
+type DBTestOptions = {
+    dbInitMethod?: 'legacy' | 'template';
+};
+
 export default async function init(
     databaseSchema = 'test',
     getLogger: LogProvider = noLoggerProvider,
-    configOverride: Partial<IUnleashOptions> = {},
+    configOverride: Partial<IUnleashOptions & DBTestOptions> = {},
 ): Promise<ITestDb> {
+    const testDbName = `unleashtestdb_${uuidv4().replace(/-/g, '')}`;
+    const useDbTemplate =
+        (configOverride.dbInitMethod ?? 'template') === 'template';
+    const testDBTemplateName = process.env.TEST_DB_TEMPLATE_NAME;
     const config = createTestConfig({
         db: {
             ...getDbConfig(),
             pool: { min: 1, max: 4 },
-            schema: databaseSchema,
+            ...(useDbTemplate
+                ? { database: testDbName }
+                : { schema: databaseSchema }),
             ssl: false,
         },
         ...configOverride,
         getLogger,
     });
 
     log.setLogLevel('error');
-    const db = createDb(config);
 
-    await db.raw(`DROP SCHEMA IF EXISTS ${config.db.schema} CASCADE`);
-    await db.raw(`CREATE SCHEMA IF NOT EXISTS ${config.db.schema}`);
-    await migrateDb(config);
-    await db.destroy();
+    if (useDbTemplate) {
+        if (!testDBTemplateName) {
+            throw new Error(
+                'TEST_DB_TEMPLATE_NAME environment variable is not set',
+            );
+        }
+        const client = new Client(getDbConfig());
+        await client.connect();
+
+        await client.query(
+            `CREATE DATABASE ${testDbName} TEMPLATE ${testDBTemplateName}`,
+        );
```

Comment:
Assumes TEST_DB_TEMPLATE_NAME should exist as it's created by jest-setup.ts script. It also assumes the db contains an already initialized and migrated db for Unleash